### PR TITLE
Switch to the new key convention

### PR DESCRIPTION
**Pull request by minad**
This change affects:

- embark-help-key
- embark-cycle-key
- embark-keymap-prompter-key

### CODE REVIEW

**Comment by oantolin** [comment]
I like the links in the customize buffer and of course, fixing my broken call to `make-obsolete`, but isn't switching from `define-key` to `keymap-set` just a loss of functionality? In particular I don't want to change the custom type from `key-sequence` to `string`.

**Comment by oantolin** [comment]
I merged the last two commits, but am waiting a bit on switching to the new key convention because it doesn't seem to have any advantages to me. If for example you told me that the plan is to eventually deprecate `define-key`, then yes, I guess we should switch.

**Comment by oantolin** [comment]
And thanks again for the fix to the make-obsolete business, I feel really bad about not testing that in fresh Emacs session where `embark-define-keymap` wasn't already defined.

**Comment by minad** [comment]
> I merged the last two commits, but am waiting a bit on switching to the new key convention because it doesn't seem to have any advantages to me.

The advantage is that the user can use kbd strings for configuration, which are clearly more user friendly and uniform than the vector or string syntax

define-key won't get deprecated since it is used internally, but from what I've seen they don't mention it anymore in the manuals.

**Comment by oantolin** [comment]
That soft deprecation is reason enough to switch for me, I guess. The custom type is a little unfortunate, but not too bad.

**Comment by oantolin** [comment]
> The advantage is that the user can use kbd strings for configuration

They already could, by using `kbd`. But it is nice to not have to type `(kbd )`. If Emacs is gradually moving in the direction of kbd strings by default that is an improvement.

**Comment by minad** [comment]
> The custom type is a little unfortunate, but not too bad.

Indeed. I guess one could specify a validation function. Maybe it is also possible to define a custom widget type or maybe Emacs already comes with such a type. I don't know.

> They already could, by using kbd. 

Yes, but that's quite ugly and I remember that key specifications were a significant source of confusion or me with Emacs for a long time.

Btw, I also moved to the new convention for the Consult key customization variables.

File: embark.el [hunk 1]
```diff
@@ -949,7 +949,11 @@ their own target finder.  See for example
 
 (defun embark--cycle-key ()
   "Return the key to use for `embark-cycle'."
-  (or embark-cycle-key (car (where-is-internal #'embark-act))))
+  (if embark-cycle-key
+      (if (key-valid-p embark-cycle-key)
+          (key-parse embark-cycle-key)
+        (error "`embark-cycle-key' is invalid"))
+    (car (where-is-internal #'embark-act))))
```
Comment (oantolin) [inline embark.el:960]: I think this is wrong: you later call `keymap-set` with this key and `where-is-internal` does not return strings that are `key-valid-p`, which is what `keymap-set` takes.

**Comment by minad** [comment]
On 2/1/23 19:19, Omar Antolín Camarena wrote:
> I think this is wrong: you later call |keymap-set| with this key and
> |where-is-internal| does not return strings that are |key-valid-p|,
> which is what |keymap-set| takes.

No, I don't think I use keymap-set with embark--cycle-key but define-key?

**Comment by oantolin** [comment]
That's right you do use `define-key`, sorry!

**Comment by oantolin** [comment]
Yes. The only really compelling argument not to merge is the custom-type and I can live with that slight loss of functionality.

**Comment by oantolin** [comment]
But I did also take the advantage to learn how to merge some but not all the commits in a PR!

**Comment by oantolin** [comment]
Fully merged now.

**Pull request merged**